x/marker/keeper/keeper.go

```diff
@@ -193,17 +193,66 @@ func (k Keeper) GetAuthority() string {
 	return k.authority
 }
 
+// IsSendDeny returns true if sender address is denied for marker
 func (k Keeper) IsSendDeny(ctx sdk.Context, markerAddr, senderAddr sdk.AccAddress) bool {
 	store := ctx.KVStore(k.storeKey)
 	return store.Has(types.DenySendKey(markerAddr, senderAddr))
 }
 
+// AddSendDeny set sender address to denied for marker
 func (k Keeper) AddSendDeny(ctx sdk.Context, markerAddr, senderAddr sdk.AccAddress) {
 	store := ctx.KVStore(k.storeKey)
 	store.Set(types.DenySendKey(markerAddr, senderAddr), []byte{})
 }
 
+// RemoveSendDeny removes sender address from marker deny list
 func (k Keeper) RemoveSendDeny(ctx sdk.Context, markerAddr, senderAddr sdk.AccAddress) {
 	store := ctx.KVStore(k.storeKey)
 	store.Delete(types.DenySendKey(markerAddr, senderAddr))
 }
+
+func (k Keeper) AddSetNetAssetValues(ctx sdk.Context, marker types.MarkerAccountI, netAssetValues []types.NetAssetValue) error {
+	for _, nav := range netAssetValues {
+		if nav.Value.Denom == marker.GetDenom() {
+			return fmt.Errorf("net asset value denom cannot match marker denom %q", marker.GetDenom())
+		}
+		if nav.Value.Denom != types.UsdDenom {
+			_, err := k.GetMarkerByDenom(ctx, nav.Value.Denom)
+			if err != nil {
+				return fmt.Errorf("net asset value denom does not exist: %v", err.Error())
+			}
+		}
+
```

Comment:
we should probably put a check in here that the volume provided does not exceed the supply of the token being set ... i.e. you can't claim a volume of `10` when the total supply of the token is `1` (for example)

Reply:
Added the check to the SetNetAssetValue method.